src/core_ocean/analysis_members/mpas_ocn_okubo_weiss.F

```diff
@@ -0,0 +1,1715 @@
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+
+      type (dm_info), intent(in) :: dminfo
+      type (block_type), intent(in) :: block
+      integer, intent(in) :: nVertLevels, nCells, nCellsSolve, nLocalCCs
+      integer, dimension(:), intent(in) :: nEdgesOnCell
+      integer, dimension(:,:), intent(in) :: cellsOnCell
+      integer, dimension(:,:), intent(in) :: OW_thresh, OW_cc_id
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      type (mpas_pool_type), pointer :: statePool
+      type (mpas_pool_type), pointer :: meshPool
+      type (mpas_pool_type), pointer :: diagnosticsPool
+
+      real (kind=RKIND), dimension(:), pointer ::  areaCell
+      real (kind=RKIND), dimension(:,:), pointer ::  layerThickness
+      real (kind=RKIND), dimension(:), pointer :: posX, posY, posZ
+      real (kind=RKIND), dimension(:,:), pointer :: velX, velY, velZ
+      logical, pointer :: on_a_sphere, use_lat_lon_coords
+
+      integer, dimension(nVertLevels*nCells) :: nextCellIdxK, nextCellIdxI
+      real (kind=RKIND), dimension(nLocalCCs) :: sumVol, numCells, origCCId, wsVelX, wsVelY, wsVelZ, &
+            wsPosX, wsPosY, wsPosZ
+      real (kind=RKIND) :: vol
+      integer k, iCell, i, curCC, iIdx, curI, curK, iOtherCell
+      integer, dimension(1) :: numCCsArray, maxNumCCsArray
+
+      integer, dimension(size(OW_thresh,1), size(OW_thresh,2)) :: OW_thresh_local
+
+      call mpas_pool_get_subpool(block % structs, 'state', statePool)
+      call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
+      call mpas_pool_get_subpool(block % structs, 'diagnostics', diagnosticsPool)
+
+      call mpas_pool_get_config(meshPool, 'on_a_sphere', on_a_sphere)
+      call mpas_pool_get_config(ocnConfigs, 'config_okubo_weiss_use_lat_lon_coords', use_lat_lon_coords)
+
+      call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, 1)
+      call mpas_pool_get_array(meshPool, 'areaCell', areaCell)
+
+      if (.not. on_a_sphere) then
+         use_lat_lon_coords = .false.
+      end if
+
+      if (use_lat_lon_coords) then
+         call mpas_pool_get_array(meshPool, 'lonCell', posX)
+         call mpas_pool_get_array(meshPool, 'latCell',  posY)
+         call mpas_pool_get_array(diagnosticsPool, 'velocityZonal', velX)
+         call mpas_pool_get_array(diagnosticsPool, 'velocityMeridional', velY)
+      else
+         call mpas_pool_get_array(meshPool, 'xCell', posX)
+         call mpas_pool_get_array(meshPool, 'yCell', posY)
+         call mpas_pool_get_array(meshPool, 'zCell', posZ)
+         call mpas_pool_get_array(diagnosticsPool, 'normalVelocityX', velX)
+         call mpas_pool_get_array(diagnosticsPool, 'normalVelocityY', velY)
+         call mpas_pool_get_array(diagnosticsPool, 'normalVelocityZ', velZ)
```

Comment:
This line, and the two above it cause issues when running a planar mesh. We don't have variables that exist named normalVelocity{X,Y,Z}. They should be renamed to velocity{X,Y,Z}.